src-v2/Judge/NTS.Judge/Adapters/Behinds/ParticipationBehind.cs

```diff
@@ -24,24 +25,46 @@ public ParticipationBehind(
 
     public IEnumerable<Participation> Participations { get; private set; } = new List<Participation>();
     public IEnumerable<IGrouping<double, Participation>> ParticipationsByDistance => Participations.GroupBy(x => x.Phases.Distance);
+    public Participation SelectedParticipation { get; private set; } = default!;
 
     // TODO: we need a better solution to load items as they have been changed in addition to load on startup.
     // Example case: importing previous data: as it is currently we have to restart the app after import
     // Maybe some sort of observable repositories?
     public async Task RunAtStartup()
     {
         Participations = await _participationRepository.ReadAll();
+        GuardHelper.ThrowIfDefault(Participations);
+        SelectParticipation(null);
     }
 
-    public async Task Process(Snapshot snapshot)
+    public void SelectParticipation(int? number)
     {
-        var participation = Participations.FirstOrDefault(x => x.Tandem.Number == snapshot.Number);
+        Participation participation = default!;
+        if (number == null)
+        {
+            participation = Participations.FirstOrDefault();
+        }
+        else
+        {
+            participation = Participations.FirstOrDefault(x => x.Tandem.Number == number);
+        }
         GuardHelper.ThrowIfDefault(participation);
+        SelectedParticipation = participation;
+        EventHelper.Emit(SelectedParticipation);
+    }
 
-        var result = participation.Process(snapshot);
+    public async Task Process(Snapshot snapshot)
+    {
+        if (SelectedParticipation == default)
+        {
+            SelectParticipation(snapshot.Number);
+        }
+        GuardHelper.ThrowIfDefault(SelectedParticipation);
+
+        var result = SelectedParticipation?.Process(snapshot);
         if (result.Type == SnapshotResultType.Applied)
         {
-            await _participationRepository.Update(participation);
+            await _participationRepository.Update(SelectedParticipation);
         }
         await _snapshotResultRepository.Create(result);
```

Comment:
Revert changes in this method as discussed.

Reply:
Actually, revert the changes in the whole file then we'll fix it using ObservableBehind after we merge all branches.